solver/app/processor.go

```diff
@@ -0,0 +1,74 @@
+package solver
+
+import (
+	"context"
+
+	"github.com/omni-network/omni/contracts/bindings"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/xchain"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// procDeps abstracts dependencies for the event processor allowed simplified testing.
+type procDeps struct {
```

Comment:
nit: this should be an interface, interface is used to do exactly what the comment says _"abstracts dependencies for ... allowed simplified testing"_

Reply:
it is a style thing, I prefer funcs, since the implementations are not related, this allows better decoupling,